src/internal/connector/graph/service.go

```diff
@@ -61,6 +78,90 @@ func (s Service) Serialize(object absser.Parsable) ([]byte, error) {
 	return writer.GetSerializedContent()
 }
 
+// ---------------------------------------------------------------------------
+// Helpers
+// ---------------------------------------------------------------------------
+
+type clientConfig struct {
+	noTimeout bool
+}
+
+type option func(*clientConfig)
+
+// populate constructs a clientConfig according to the provided options.
+func (c *clientConfig) populate(opts ...option) *clientConfig {
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	return c
+}
+
+// apply updates the http.Client with the expected options.
+func (c *clientConfig) apply(hc *http.Client) {
```

Comment:
Add a small unit test for this please. Let's validate that apply does the right thing for the http client.

Reply:
Can do.